legacy/c3_trajectory_generator/src/waypoint_validity.cpp

```diff
@@ -70,9 +75,32 @@ std::pair<bool, WAYPOINT_ERROR_TYPE> WaypointValidity::is_waypoint_valid(const g
   return std::make_pair(true, WAYPOINT_ERROR_TYPE::UNOCCUPIED);
 }
 
+void WaypointValidity::pub_size_ogrid(const geometry_msgs::Pose &waypoint, int d)
+{
+  std::vector<int8_t> sub_ogrid_data(std::pow(sub_ogrid_size_ / ogrid_map_->info.resolution, 2), d);
+  nav_msgs::OccupancyGrid rosGrid;
+  rosGrid.header.seq = 0;
+  rosGrid.info.resolution = ogrid_map_->info.resolution;
+  rosGrid.header.frame_id = "map";
+  rosGrid.header.stamp = ros::Time::now();
+  rosGrid.info.map_load_time = ros::Time::now();
+  rosGrid.info.width = sub_ogrid_size_ / ogrid_map_->info.resolution;
+  rosGrid.info.height = sub_ogrid_size_ / ogrid_map_->info.resolution;
+  rosGrid.info.origin.position.x = waypoint.position.x - sub_ogrid_size_ / 2;
+  rosGrid.info.origin.position.y = waypoint.position.y - sub_ogrid_size_ / 2;
+  rosGrid.data = sub_ogrid_data;
+  if (d == 200)
```

Comment:
You shouldn't branch on arbitrary magic constants like this. Consider using a flag parameter.